### PR TITLE
Continue Refactoring Counts

### PR DESCRIPTION
**Pull request by droogmic**

### CODE REVIEW

File: osm2lanes/src/transform/tags_to_lanes/road.rs [hunk 1]
```diff
@@ -204,46 +191,80 @@ impl RoadBuilder {
...
+            })
+            .take(backward.some().unwrap_or(0))
+            .collect();
 
-        let road = RoadBuilder {
-            forward_lanes,
-            backward_lanes,
-            highway,
-            oneway,
+            // TODO Fix upstream. https://wiki.openstreetmap.org/wiki/Key:centre_turn_lane
```
Comment (dabreegster) [inline osm2lanes/src/transform/tags_to_lanes/road.rs:236]: Old comment, can delete. This was me thinking that I ought to go fix all remaining uses of this myself...

File: osm2lanes/src/transform/tags_to_lanes/road.rs [hunk 2]
```diff
@@ -204,46 +191,80 @@ impl RoadBuilder {
...
+            }
+
+            RoadBuilder {
+                forward_lanes,
+                backward_lanes,
+                highway,
+                oneway,
+            }
+        } else {
+            RoadBuilder {
```
Comment (dabreegster) [inline osm2lanes/src/transform/tags_to_lanes/road.rs:254]: Nice, much cleaner to split these cases.

FYI / aside: on the A/B Street side, I have to turn around and map most cases of Direction::Both to two directional lanes. One day I'd like to "natively" handle bidi lanes at the simulation layer and elsewhere...

**Review by dabreegster — APPROVED** [review]

File: osm2lanes/src/transform/tags_to_lanes/road.rs (hunk 1 above)
Comment (droogmic) [reply osm2lanes/src/transform/tags_to_lanes/road.rs:236]: haha

```suggestion
```

File: osm2lanes/src/transform/tags_to_lanes/road.rs (hunk 2 above)
Comment (droogmic) [reply osm2lanes/src/transform/tags_to_lanes/road.rs:254]: yeah, I understand.

**Pull request merged**